File: Java/Medium/WrongAnswer/all-ancestors-of-a-node-in-a-directed-acyclic-graph_0.java
class Solution {
    public List<List<Integer>> getAncestors(int n, int[][] edges) {
        
        List<List<Integer>> graph = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            graph.add(new ArrayList<>());
        }
        for (int[] edge : edges) {
            graph.get(edge[1]).add(edge[0]);
        }
        List<List<Integer>> res = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            Set<Integer> ancestors = new HashSet<>();
            dfs(graph, i, ancestors);
            res.add(new ArrayList<>(ancestors));
        }
        return res;
    }
    private void dfs(List<List<Integer>> graph, int i, Set<Integer> ancestors) {
        for (int j : graph.get(i)) {
            ancestors.add(j);
            dfs(graph, j, ancestors);
        }
    }
}

File: Rust/Medium/WrongAnswer/all-ancestors-of-a-node-in-a-directed-acyclic-graph_0.rs
impl Solution {
    pub fn get_ancestors(n: i32, edges: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        
        let mut graph = vec![vec![]; n as usize];
        for edge in edges {
            graph[edge[1] as usize].push(edge[0]);
        }
        let mut result = vec![vec![]; n as usize];
        for i in 0..n {
            let mut queue = vec![];
            queue.push(i);
            while !queue.is_empty() {
                let node = queue.pop().unwrap();
                for neighbor in &graph[node as usize] {
                    if !result[i as usize].contains(neighbor) {
                        result[i as usize].push(*neighbor);
                        queue.push(*neighbor);
                    }
                }
            }
        }
        result
    }
}